pkg/bootstrap/platform/azure.go

```diff
@@ -0,0 +1,170 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package platform
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"sync"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+
+	"istio.io/pkg/log"
+)
+
+const (
+	AzureMetadataEndpoint = "http://169.254.169.254"
+	InstanceURL           = AzureMetadataEndpoint + "/metadata/instance"
+	DefaultAPIVersion     = "2019-08-15"
+	SysVendorPath         = "/sys/class/dmi/id/sys_vendor"
+	MicrosoftIdentifier   = "Microsoft Corporation"
+)
+
+var (
+	APIVersion        = DefaultAPIVersion
+	updateVersionOnce = sync.Once{}
+	getAPIVersions    = func() string {
```

Comment:
nit: go tends to not use the `get` prefix. since this package is shared by multiple platforms, maybe we can rename as follows:

```golang
azureAPIVersionsFn = ...
azureMetadataFn = ...
```